### PR TITLE
Address PR #515 review follow-ups

### PR DESCRIPTION
**Pull request by leynos**
## Summary

Works through the code-review comments left on #515 (Harden workspace lint
configuration) after it merged. Every finding was verified against the current
tree before any change; one was found to be stale and is left alone with a
reason below.

## Correctness and isolation

- **BDD IR steps read the scenario environment.** `compile_manifest_impl` called
  `manifest::from_path`, bypassing the injected environment, so IR scenarios read
  host variables while the manifest scenarios read scenario values. It now uses
  `from_path_with_policy_and_env` with the same `manifest_env_reader` seam.
- **`tests/cli_tests/merge.rs` now sandboxes both merge sites.** `with_config_file`
  and the precedence test called `merge_in_child` directly, letting host XDG
  configuration reach the child. Both now go through `isolated_environment`.
- **Merge-probe allowlist forwards the dynamic-loader paths.** `LD_LIBRARY_PATH`
  and `DYLD_FALLBACK_LIBRARY_PATH` are Cargo-supplied; clearing them leaves the
  re-executed worker unable to start on Linux and macOS.
- **Project-scope discovery gets its own home directory** instead of aliasing the
  project directory, so the assertions can only pass through project scope.
- **`UndefinedError` is asserted before the downcast** in the
  missing-environment-variable manifest test, so the cause is checked whatever
  the outer error type; the `ManifestError::Parse` check is retained for wrapped
  values.
- **Explicit-config assertion pinned to the stable diagnostic** as well as the
  missing path, so a generic I/O failure cannot satisfy it.
- **Whitaker boundary contract parses arguments.** Substring matching could not
  tell `--package test_support` from `--package test_support-extra`, nor spot a
  second selection; a small tokenizer now requires exactly one selection per
  invocation.
- **Workflow YAML root validated structurally.** `_load` pattern-matches the
  `yaml.safe_load` result and rejects scalar, list, and `None` roots. Note that
  GitHub Actions' `on:` key is a YAML 1.1 boolean word, so PyYAML returns it as
  `True`; the key is restored to `"on"` before the string-key check.
- **Non-UTF-8 ninja override fails loudly** rather than converting lossily, and
  the packaging assertion now names the offending path.
- **`ENV_LOCK` poison flag cleared before asserting**, so a failing assertion
  cannot leak process-global poisoning; the test carries a note that its direct
  global-mutex probe depends on nextest's per-test process isolation.

## Structure and documentation

- Replaced the `cfg`-gated bare `return` in `EnvSnapshot::capture_with_env` with
  per-platform wrappers over the two divergent `capture_impl` arities.
- Kept the compiler-wrapper marker sidecar in camino's UTF-8 domain, converting
  once at the `std::path` boundary.
- Split the installation, release-detail, and Windows setup contracts out of
  `tests/documentation_examples_tests.rs` (495 lines) into
  `tests/documentation_examples_installation_tests.rs`, following the existing
  convention of one test binary per concern sharing the `documentation_examples`
  loader module. The original file is now 354 lines.
- Corrected `initialise`/`initialisation` to Oxford spelling in the macro
  initialization message (both English locales) and the fetch-cache Rustdoc,
  updating the inline `insta` snapshot in lockstep.
- Documented the native-path contract for the injected Ninja executable, and
  filled the Rustdoc gaps in \`fake_ninja_check_build_file_in\`,
  \`real_utility_with_env\`, \`stdlib_output_or_error\`, and the non-Unix
  \`make_executable\` (whose \`#[cfg]\` also sat inside its doc comment).

## Not actioned

One comment asked to rewrite the \`run_netsuke_in\` environment-contract comment
to say no process environment variables are inherited. That helper does not call
\`env_clear\` — it inherits the parent environment except for the selectors it
explicitly sets or removes — so the existing comment is accurate and the
suggested wording describes \`run_netsuke_in_with_env\` instead. Applying it
would have made the comment wrong.

## Validation

All deterministic commit gates pass on a clean tree: \`check-fmt\`, \`lint\`
(rustdoc, Clippy with \`-D warnings\`, Whitaker for both crates), \`typecheck\`,
\`test\` (1652 nextest tests, 83 doctests), \`markdownlint\`, \`nixie\`, and
\`test-workflow-contracts\` (12/12).

## References

- Follows up on #515 — Harden workspace lint configuration
- Lody session: https://lody.ai/leynos/sessions/44745a2f-e479-493e-95e3-4862edefc48d

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Address post-merge review feedback to harden lint-related tests, environment handling, and documentation contracts across manifests, workflows, packaging, and installation flows.

Bug Fixes:
- Ensure BDD IR compilation and manifest steps both read the injected scenario environment instead of the host environment.
- Sandbox CLI merge tests so child processes cannot see host XDG configuration and preserve dynamic loader paths needed to start re-executed workers.
- Tighten Whitaker boundary tests to require exactly one package selection per invocation and avoid substring-based false positives.
- Validate workflow YAML roots structurally as string-keyed mappings to surface malformed documents with clear assertions.
- Fail loudly on non-UTF-8 Ninja override paths instead of performing lossy conversions, and improve packaging assertions by naming offending manifest entries.
- Prevent global ENV_LOCK poisoning from leaking between tests by clearing the poison flag before asserting state.
- Assert MiniJinja UndefinedError causes for missing env vars regardless of the outer error type, and pin explicit-config error diagnostics to their stable messages.
- Ensure project-scope config discovery tests use a distinct home directory so they only pass via the intended scope.

Enhancements:
- Refactor EnvSnapshot environment capture into platform-specific wrappers to avoid cfg-gated early returns and clarify PATHEXT handling.
- Document and re-export test helpers for resolving real utilities via injected environments and clarify behaviour of stdlib_output_or_error, fake Ninja helpers, and make_executable on different platforms.
- Keep compiler-wrapper marker paths within camino's UTF-8 domain by converting at the std::path boundary and tightening associated tests.
- Adjust manifest macro error messaging and documentation to consistent spelling and update associated snapshots.
- Expose manifest environment reader helpers for reuse and add documentation clarifying scenario-based environment handling.

Documentation:
- Split installation and release documentation example tests into a dedicated binary to keep documentation_examples_tests within size limits while preserving installation and Windows setup contracts.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai resume

File: test_support/src/env_lock.rs [hunk 1]
```diff
@@ -246,15 +252,13 @@ mod tests {
         });
 
         assert!(poisoner.join().is_err(), "poisoning thread should panic");
-        assert!(
-            ENV_LOCK.is_poisoned(),
-            "the panic should poison the underlying mutex"
-        );
+        let was_poisoned = ENV_LOCK.is_poisoned();
+        ENV_LOCK.clear_poison();
+        assert!(was_poisoned, "the panic should poison the underlying mutex");
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/env_lock.rs:280]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: test_support/src/http/tests.rs [hunk 2]
```diff
@@ -56,7 +56,11 @@ struct DurationCase {
     key: &'static str,
     value: Option<&'static str>,
     expected: Duration,
-    expected_warning_value: Option<&'static str>,
+    /// Bounded parse-failure text expected in the warning, if it should warn.
+    ///
+    /// Deliberately not the offending value: the warning redacts it, so
+    /// asserting on a category is what keeps that redaction honest.
+    expected_warning_error: Option<&'static str>,
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/http/tests.rs:63]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/packaging_smoke_tests.rs [hunk 3]
```diff
@@ -95,14 +95,18 @@ fn assert_required_paths_present(packaged_paths: &BTreeSet<&str>) {
...
-            packaged_paths.iter().all(|path| {
-                Path::new(path)
-                    .components()
-                    .next()
-                    .is_none_or(|component| component.as_os_str() != forbidden_root)
-            }),
-            "packaged manifest should not contain `{forbidden_root}`"
+            offender.is_none(),
+            "packaged manifest should not contain `{forbidden_root}`, found `{}`",
+            offender.copied().unwrap_or_default()
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/packaging_smoke_tests.rs:112]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/workflow_contracts/ci_lint_test.py [hunk 4]
```diff
@@ -39,8 +39,34 @@
...
+    GitHub Actions' ``on:`` trigger key is a YAML 1.1 boolean word, so PyYAML
+    hands it back as ``True``; it is restored to ``"on"`` so callers see a
+    uniformly string-keyed mapping.
+    """
+    match yaml.safe_load(WORKFLOW_PATH.read_text(encoding="utf-8")):
+        case dict() as document:
+            workflow = {
+                ("on" if key is True else key): value
+                for key, value in document.items()
+            }
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/workflow_contracts/ci_lint_test.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Thanks — I've actioned part of the **Unit Architecture** finding and am declining the rest, with reasoning below. I'm also declining the **User-Facing Documentation** finding on factual grounds.

## Unit Architecture — partially actioned

You're right that `render_template` was reaching for the global metric registry itself: it called `describe_render_metrics()` in its own body. That moved into `instrument_template_render` (c516531), so the query function now names only the instrumentation boundary it composes with. The macro-side registration stays in `make_macro_fn`, because that runs when a macro is *registered* — setup, not evaluation — so it never sits on the invocation path.

I am **not** injecting a clock and telemetry sink. This reverses no new information since ADR-008 recorded the decision, and four specific things weigh against it:

1. **The project's observability policy endorses the current pattern.** `AGENTS.md`: "Libraries may emit `metrics` and `tracing` instrumentation, but must not install global recorders or subscribers." Emitting from a library through the facade *is* the mandated shape; the prohibition is on installing recorders, which this module does not do.

2. **The injection seam already exists, and the tests already use it.** The goal — exercise this code without global state — is met today via `metrics::with_local_recorder` (scoped, not global) plus a scoped `tracing` subscriber. See `src/manifest/tests/macros_telemetry.rs` and `src/manifest/tests/macro_invocation_telemetry.rs`. A bespoke sink trait would duplicate the abstraction `metrics` exists to provide, while still needing the same redaction discipline.

3. **The "outer adapter" alternative is structurally impossible at the macro granularity.** `make_macro_fn` returns a callback that **MiniJinja** invokes during evaluation; there is no outer call site at per-invocation granularity to hoist instrumentation to. The only enclosing boundary is `render_template`, which is a different granularity — moving it there would collapse the two boundaries into one and lose the render-versus-fallback split. That split is documented in ADR-008 and pinned by `imported_macro_render_does_not_emit_invocation_metrics`, so collapsing it would be a capability regression, not a refactor.

4. **The cost lands on the public API.** The macro callback is constructed via `register_macro` → `register_manifest_macros` → `from_str_named`, so a sink and clock would have to be threaded through five public functions — `from_str`, `from_str_with_env`, `from_path`, `from_path_with_policy`, and `from_path_with_policy_and_env` — putting observability plumbing into the manifest-loading API to serve a seam that already exists.

ADR-008 records this decision, including an explicit "Inject a telemetry-sink trait — Rejected" entry with rationale, and is referenced from `docs/netsuke-design.md`. Reversing it would need new information rather than a restatement.

## User-Facing Documentation — declining

The resolution asks for a `docs/users-guide.md` section listing the metrics and **"exporter setup requirements"**. Those requirements do not exist: there is no `set_global_recorder`, `install_recorder`, or exporter anywhere in `src/`, and `metrics-util` is a dev-dependency only. Every metric currently emits into the `metrics` crate's no-op global recorder for CLI users, so there is nothing for an operator to configure, and documenting setup steps would describe a capability the tool does not have.

The only audience that can observe these metrics is a consumer embedding `netsuke-build` as a library and installing its own recorder — integrator territory, which is where the contract is documented: the developers' guide section "Manifest telemetry: template render and macro invocation", plus ADR-008. This also matches the repository's documentation-scope boundary, which keeps contributor- and integrator-facing material out of the users' guide.

If an exporter is ever installed in the binary, that is exactly when a users'-guide section becomes correct, and I'd add it then.

## Also in this round

Both inline comments were valid and are fixed:

- `test_support/src/env_lock.rs` — the safety note claimed *every* assertion runs with the poison flag cleared. That was wrong for the `join` assertion, which precedes `clear_poison()`; it is safe for a different reason (the poisoned guard belonged to the spawned thread, so this thread has no live guard to drop). The note now separates the two cases (427b504).
- `docs/adr-008-*.md` — removed the comma before "because" in both sentences. The second needed more than the comma: "covers the compiled-expression fallback only because imports evaluate" reads as "merely because", so `only` moved ahead of the noun phrase to preserve the meaning.

One follow-up spotted while checking the docs: moving `describe_render_metrics` had left the developers' guide asserting it "runs at the top of every `render_template` call". Corrected in e14efb4.

All gates green: `check-fmt`, `lint` (Clippy + Whitaker, both crates), `typecheck`, 1668 nextest tests, 84 doctests, `markdownlint`, `nixie`, and `test-workflow-contracts`.

File: docs/adr-008-bounded-redacted-manifest-telemetry.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+Manifest evaluation renders Jinja templates and invokes manifest-defined
+macros. Both are queries: they compute a value and were expected to stay free
+of ambient concerns such as timing and metric emission. Once observability was
+needed for these paths, two risks had to be weighed against each other:
+
+- Manifest content — template text, macro names, macro arguments, and context
+  values — is caller-controlled and unbounded. Recording it directly in a
+  metric label produces unbounded cardinality in the metric series, and
+  recording it in a trace risks leaking secrets, because environment variable
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/adr-008-bounded-redacted-manifest-telemetry.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by buzzybee-df12** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai please create a new GitHub issue for the following:

### ❌ Failed checks (1 error)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Telemetry, HTTP redaction, IR environment, path, and lock changes have substantive tests. However, the new `_WorkflowLoader` and `_load` validation have no tests for their YAML branches; existing t... | Add focused loader tests that assert `on` remains a string, true/false resolve, and `_load` rejects empty, scalar, list, and non-string-key mappings. |

File: tests/novice_flow_smoke_tests.rs [hunk 6]
```diff
@@ -1,8 +1,10 @@
 //! Smoke tests for newcomer-facing CLI flows.
 
-use anyhow::{Context, Result, ensure};
+use anyhow::{Context, Result, bail, ensure};
 use rstest::rstest;
 use std::path::Path;
+#[cfg(unix)]
+use std::path::PathBuf;
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/novice_flow_smoke_tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai resume
